File: cpp/rna-transcription/rna_transcription.cpp
#include "rna_transcription.h"
#include <algorithm>

using namespace std;

namespace transcription {

char to_rna(char unit) {
  char ret = '?';
  switch (unit) {
  case 'C':
    ret = 'G';
    break;
  case 'G':
    ret = 'C';
    break;
  case 'A':
    ret = 'U';
    break;
  case 'T':
    ret = 'A';
    break;
  }
  return ret;
}

string to_rna(string strand) {
  for (auto it=begin(strand); it!=end(strand); ++it) {
    *it = to_rna(*it);
  }
  return strand;
}

}

File: go/rna-transcription/rna_transcription.go
package strand

import "strings"

// TestVersion indicates which exercism test suite to use
const TestVersion = 1

// ToRna coverts an input DNA strand to its RNA complement.
func ToRna(dna string) string {
	rnaTranscription := func(nucleotide rune) rune {
		switch nucleotide {
		case 'G':
			return 'C'
		case 'C':
			return 'G'
		case 'T':
			return 'A'
		case 'A':
			return 'U'
		}
		panic("Invalid DNA sequence provided... Input: " + dna)
	}
	return strings.Map(rnaTranscription, dna)
}
